docs/architecture/adr-006-poap-v2.md

```diff
@@ -0,0 +1,75 @@
+# ADR 006: POAP v2 Contract
+
+## Changelog
+
+- Jul 11, 2023: Initial draft;
+
+## Status
+PROPOSED Not Implemented
+
+## Abstract
+This ADR defines the new architecture of a POAP v2 Smart Contract that will replace the current `cw721-poap`, `poap-manager`, and `poap` smart contracts.
+
+## Context
+The name POAP stands for Proof of Attendance Protocol, which is a simple NFT that attests to the fact that a user has participated in a particular event.
+
+Currently, we have implemented the ability to create and manage POAPs with three smart contracts: `cw721-poap`, `poap-manager`, and `poap`. Although this division works and makes sense under the Single Responsibility Principle, there are several downsides to it. First, users can easily upload and instantiate multiple instances of each of those contracts, leading to an unnecessary amount of contracts on the chain. Second, it makes it harder for newcomers to understand the interactions between the three contracts. Lastly, having multiple smart contracts for something as simple as a POAP seems excessive, as ideally, a POAP should have minimal operations.
+
+## Decision
+The decision is to merge all POAP-related operations into a single smart contract called `poap`, replacing all existing POAP-related smart contracts.
+
+## Specifications
+Below are the specifications for the various messages that the `poap` contract should support.
+
+### Messages
+
+#### Instantiate
+To instantiate this smart contract, the user needs to provide:
+
+* `admin` address: Identifies the user that controls the contract.
+* `poap_name`: Identifies the name of the minted NFTs.
+* `poap_symbol`: Identifies the symbol of the minted NFTs.
+* `poap_metadata_uri`: Identifies the URI where users can view the associated metadata for the POAPs, ideally following the ERC-721 metadata scheme in a JSON file.
+* `poap_mint_limit_per_address`: Specifies the maximum number of POAPs that can be minted per address.
+* `poap_mint_enabled`: Indicates whether users can mint the POAPs.
+* `poap_is_transferable`: Specifies whether each POAP can be transferred from one user to another.
+* `event_start_date`: Identifies the start date of the event associated with the POAP.
+* `event_end_date`: Identifies the end date of the event associated with the POAP.
```

Comment:
Are those `event_start_date` and `event_end_date` two datetimes (timestamps)?
Are they used to boundary the minting in time? If this, should they be called `minting_start_date` and `minting_end_date`?
Otherwise, should they be inserted in the metadata?

Reply:
Yes, these actually determine the timeframe during which the POAP will be mintable. I have now updated their names to be `poap_mint_start_time` and `poap_mint_end_time` to be easier to understand.